content/courses/Guidelines_for_Safe_and_Secure_Ada_SPARK/chapters/guidelines/exception_usage/exu02_no_unhandled_application-defined_exceptions.rst

```diff
@@ -0,0 +1,111 @@
+-----------------------------------------------------
+No Unhandled Application-Defined Exceptions (EXU02)
+-----------------------------------------------------
+
+.. include:: ../../../global.txt
+
+**Level** :math:`\rightarrow` Required
+
+Category
+   :Safety: :math:`\checkmark`
+   :Cyber: :math:`\checkmark`
+
+Goal
+   :Maintainability: :math:`\checkmark`
+   :Reliability: :math:`\checkmark`
+   :Portability: :math:`\checkmark`
+   :Performance:
+   :Security:
+
+**Remediation** :math:`\rightarrow` Low
+
+**Verification Method** :math:`\rightarrow` GNATcheck rule:
+:rule:`Unhandled_Exceptions`
+
++++++++++++
+Reference
++++++++++++
+
+N/A
+
++++++++++++++
+Description
++++++++++++++
+
+All application-defined exceptions must have at least one corresponding handler
+that is applicable. Otherwise, if an exception is raised, undesirable behavior
+is possible. The term *applicable* means that there is no dynamic call chain
+that can reach the active exception which does not also include a handler that
+will be invoked for that exception, somewhere in that chain.
+
+When an unhandled exception occurrence is raised in the sequence of statements
```

Comment:
When an unhandled exception occurs in the sequence of statements of an application task and propagates to the task's body, the task terminates abnormally.

Reply:
Fixed